tests/test_secs.py

```diff
@@ -242,3 +258,230 @@ def test_list_numpy():
     assert secs_list2.nsec == secs_np2.nsec
     assert_array_equal(secs_list2.sec_df_loc, secs_np2.sec_df_loc)
     assert_array_equal(secs_list2.sec_cf_loc, secs_np2.sec_cf_loc)
+
+
+def test_sec_bad_shape():
+    """Test bad input shape."""
+    # Wrong dimensions
+    x = np.array([[1, 0], [1, 0]])
+    with pytest.raises(ValueError, match='SEC DF locations'):
+        pysecs.SECS(sec_df_loc=x)
+    with pytest.raises(ValueError, match='SEC CF locations'):
+        pysecs.SECS(sec_cf_loc=x)
+
+
+def test_one_sec():
+    """Test 1-dimensional input location gets mapped properly."""
+    x1d = np.array([1, 0, 0])
+    secs = pysecs.SECS(sec_df_loc=x1d)
+    assert secs.nsec == 1
+    secs = pysecs.SECS(sec_cf_loc=x1d)
+    assert secs.nsec == 1
+
+
+def test_fit_unit_currents():
+    """Test the unit current function"""
+    # divergence free
+    secs = pysecs.SECS(sec_df_loc=[[1., 0., 0.], [1., 0., 0.]])
+    secs.fit_unit_currents()
+    assert_array_equal(np.ones((1, 2)), secs.sec_amps)
+
+    # curl free
+    secs = pysecs.SECS(sec_cf_loc=[[1., 0., 0.], [1., 0., 0.]])
+    secs.fit_unit_currents()
+    assert_array_equal(np.ones((1, 2)), secs.sec_amps)
+
+    # divergence free + curl free
+    secs = pysecs.SECS(sec_df_loc=[[1., 0., 0.], [1., 0., 0.]],
+                       sec_cf_loc=[[1., 0., 0.], [1., 0., 0.]])
+    secs.fit_unit_currents()
+    assert_array_equal(np.ones((1, 4)), secs.sec_amps)
+
+
+def test_fit_bad_obsloc():
+    """Test bad observation locations input."""
+    secs = pysecs.SECS(sec_df_loc=[1., 0., R_EARTH + 1e6])
+    obs_loc = np.array([[0, 0]])
+    obs_B = np.array([[1, 1, 1]])
+    with pytest.raises(ValueError, match="Observation locations"):
+        secs.fit(obs_loc, obs_B)
+
+
+def test_fit_one_time():
+    """One timestep fit"""
+    secs = pysecs.SECS(sec_df_loc=[[1., 0., R_EARTH + 1e6],
+                                   [-1., 0., R_EARTH + 1e6]])
+    obs_loc = np.array([[0, 0, R_EARTH]])
+    obs_B = np.array([[1, 1, 1]])
+    secs.fit(obs_loc, obs_B)
+    assert_allclose([[6.40594202e+13, -7.41421248e+13]], secs.sec_amps)
+
+
+def test_fit_multi_time():
+    """One timestep fit"""
+    secs = pysecs.SECS(sec_df_loc=[[1., 0., R_EARTH + 1e6],
+                                   [-1., 0., R_EARTH + 1e6]])
+    obs_loc = np.array([[0, 0, R_EARTH]])
+    obs_B = np.ones((2, 1, 3))
+    obs_B[1, :, :] *= 2
+    secs.fit(obs_loc, obs_B)
+    arr = np.array([6.40594202e+13, -7.41421248e+13])
+    expected = np.array([arr, 2*arr])
+    assert_allclose(expected, secs.sec_amps)
+
+
+def test_fit_obs_var():
+    """One timestep fit"""
```

Comment:
Trivial, but isn't this a two timestep fit?

Reply:
copy-paste error from above, thanks for catching that! I'll go through them all and try to fix them up.